### PR TITLE
Add support for a border between sidebar and panel titles and views

### LINKED ISSUE #157312: Add support for borders below Sidebar and Panel titles.
Hi,

I think it would be useful to add support for a border below sidebar and panel titles.

Eg, current Light High Contrast Theme:

image

It looks a bit odd how the the borders between panels just goes up and meets nothing.

Proposed:

image

I think this border should default to null so be a no-op for regular themes, although, you could make a case for setting it for high contrast themes. My main ask is just to have it as something that can be configured by themes.

### PR DESCRIPTION
**Pull request by a-stewart**
This fixes #157312

Before:  
image

After:  
image

### CODE REVIEW

**Comment by a-stewart** [comment]
Hi,

Small ping on this. I'm happy to update the PR with the newest changes if this is something you would be willing to consider?

I think it gives more flexibility to theme owners and makes sense from a UI point of view with the high contrast themes.

**Comment by benibenj** [comment]
Apologies for the delay here, ownership of this feature area has recently changed. 
This seems reasonable to me. I see there are a couple of merge conflicts going on. If you fix those I'd be happy to have a look at the PR.

**Comment by a-stewart** [comment]
> Apologies for the delay here, ownership of this feature area has recently changed.
This seems reasonable to me. I see there are a couple of merge conflicts going on. If you fix those I'd be happy to have a look at the PR.

No worries, I've merged in the latest changes and resolved the issues.

File: src/vs/workbench/browser/parts/compositePart.ts [hunk 1]
```diff
@@ -436,6 +437,8 @@ export abstract class CompositePart<T extends Composite> extends Part {
 
 			updateStyles: () => {
 				titleLabel.style.color = $this.titleForegroundColor ? $this.getColor($this.titleForegroundColor) || '' : '';
+				const border = $this.titleBorderColor ? $this.getColor($this.titleBorderColor) : undefined;
+				parent.style.borderBottom = border ? `1px solid ${border}` : '';
```
Comment (benibenj) [inline src/vs/workbench/browser/parts/compositePart.ts:441]: Just setting the border bottom will take away 1px from the title height causing everything to not be centered anymore. We might need to solve this the same way we solve it for editor title (tab bar). We use the ::after element in that case.

See here:
https://github.com/a-stewart/vscode/blob/6b8908833b519af7b37fee9772c6af205dc4fa80/src/vs/workbench/browser/parts/editor/editorGroupView.ts#L2141-L2149
https://github.com/a-stewart/vscode/blob/6b8908833b519af7b37fee9772c6af205dc4fa80/src/vs/workbench/browser/parts/editor/media/editorgroupview.css#L121-L131

**Review by benibenj — CHANGES_REQUESTED** [review]

**Comment by benibenj** [comment]
image

**Comment by a-stewart** [comment]
Just done some more experimenting and setting the height to anything else causes issues too.

Additionally, I think we want the border line where it is so that everything lines up in the top row.

Looking at this everything should work, except that the `.componsite-bar` which normally has height 100% is fixed at 35px in the `.title`. Removing this hard coding seems to work as expected.

File: src/vs/workbench/browser/parts/compositePart.ts (hunk 1 above)
Comment (a-stewart) [reply src/vs/workbench/browser/parts/compositePart.ts]: Resolving this conversation - the approach above wouldn't work but have adjusted the code to handle the centering.

**Review by benibenj — APPROVED** [review]

**Review by lszomoru — APPROVED** [review]

**Comment by vimchun** [comment]
Hi, now, I have a lot of borders in green everywhere on VSCode with my theme "One Dark Pro".
I know I can put a border in transparent, but I think there are a lot.
I would like to have all these new border transparent, or not displaying at all (like it was before the update).
Is there an option to do it, instead of doing something like that ?

```
{
    "workbench.colorCustomizations": {
        "[One Dark Pro]": {
            "titleBar.border" : "#00000000",
            "activityBar.border" : "#00000000",
            "statusBar.border": "#00000000",
            "sideBar.border":  "#00000000",
            "input.border": "#00000000",
            "panel.border" : "#00000000"
            // ...
            // ...
        }
    }
}
```

**Comment by a-stewart** [comment]
Unless something weird has happened with the merge this should default to undefined.

I've tested One Dark Pro (the one with 10 million installs) on insiders.vscode.dev and it looks normal.

Could you confirm exactly which theme you are using from which extension please.

**Comment by benibenj** [comment]
@vimchun can you also specify if you changes the activity bar location? Did you maybe move it to the top? It might look different then

**Comment by a-stewart** [comment]
It seems like on the non-web version setting `"oneDarkPro.editorTheme": "Retro"` will add borders (after a reload), but it looks like this is the desired effect?

**Comment by vimchun** [comment]
@benibenj : 
No, my activity bar is on the bottom. 

@a-stewart 
I use this theme : https://github.com/Binaryify/OneDark-Pro
Thank you, you are right, I recently added `"oneDarkPro.editorTheme": "Retro"`. 
When I saw the issue (which is not an issue according to the screenshot on the theme github), i tried to reproduce it with a minimum configuration, but did not find the culprit, because I did not reload VSCode (I knew something must be wrong with my configuration, as I did not see anyone else complaining).
After removing that setting, everything is fine (I don't have these green lines everywhere).

Thanks for your help !

**Comment by vimchun** [comment]
@benibenj :
My activity bar is on the right side.

@a-stewart
I use this theme : https://github.com/Binaryify/OneDark-Pro
You are right, I recently added `"oneDarkPro.editorTheme": "Retro"`.
I could not find it was the culprit on my configuration, because I did not reload VSCode after removing it (I knew something must be wrong with my configuration, as I did not see anyone else complaining).
After removing that setting (and reload VSCode), everything is fine.

Thank you for your help !

**Pull request merged**